fmralign/examples/pairwise_roi_alignment.py

```diff
@@ -0,0 +1,168 @@
+                    roi_masker.transform(im_train_2))
+# Predict test data for subject 2 from subject 1
+predicted_data = alignment_class.transform(roi_masker.transform(im_test_1))
+
+#############################################################################
+# Score the prediction of test data without alignment
+# ---------------------------------------------------
+# To score the quality of prediction we use r2 score \
+#   on each voxel activation profile across contrasts
+# This score is 1 for a perfect prediction and can get \
+#   arbitrarly bad (here we clip it to -1 for bad predictions)
+
+import numpy as np
+from sklearn.metrics import r2_score
+# Mask the real test data for subject 2 to get a ground truth vector
+ground_truth = roi_masker.transform(im_test_2)
+
+# Score the prediction of test data without alignment...
+baseline_score = np.maximum(
+    r2_score(ground_truth, roi_masker.transform(im_test_1),
+             multioutput='raw_values'), -1)
+# ... and using alignment.
+aligned_score = np.maximum(
+    r2_score(ground_truth, predicted_data,
+             multioutput='raw_values'), -1)
+
+#############################################################################
+# Plotting the prediction quality
+# --------------------------------
+#
+from nilearn.plotting import plot_stat_map
+baseline_display = plot_stat_map(roi_masker.inverse_transform(
+    baseline_score), display_mode="z", cut_coords=[-15, -5], vmax=0.5)
+baseline_display.title("R2 score between raw data")
+display = plot_stat_map(
+    roi_masker.inverse_transform(
+        aligned_score), display_mode="z", cut_coords=[-15, -5], vmax=0.5)
+display.title("R2 score after alignment")
+#############################################################################
+# We can see on the plot that after alignment, the prediction made for one
+# subject data, informed by another subject are greatly improved.
+#############################################################################
+# Aligning on ROI directly with PairwiseAlignment
+# ---------------------------------------------------
+# Instead of masking the data and applying alignment separately, /
+# we could also be have done the same directly using PairwiseAlignment() /
+# with the visual mask, on nifti images.
+from fmralign.pairwise_alignment import PairwiseAlignment
+
+alignment_estimator = PairwiseAlignment(
+    alignment_method='scaled_orthogonal', n_pieces=1, mask=roi_masker)
+alignment_estimator.fit(im_train_1, im_train_2)
+directly_predicted_img = alignment_estimator.transform(im_test_1)
+directly_aligned_score = np.maximum(
+    r2_score(ground_truth, roi_masker.transform(directly_predicted_img),
+             multioutput='raw_values'), -1)
+display = plot_stat_map(roi_masker.inverse_transform(directly_aligned_score),
+                        display_mode="z", cut_coords=[-15, -5], vmax=0.5)
+display.title("R2 score after alignment (using PairwiseAlignment)")
```

Comment:
I don't really see the added value of the repetition. You should use only the class that  you think users should use.
When there are two different ways to get to the same result, this creates confusions.

Reply:
Do you think that users need to know how to use the classes on array to be independent of our masking process and all ? That's why I framed the example that way in the first place.

But the overall API I would wish to enforce would be to use only Pairwise / Template alignment classes that internalize masking and classe use. I'm not sure it's flexible enough for every need though.

Reply:
But then let's try and enforce this. if for some reasons this is not sufficient, you'll figure it out, or users  will --- for sure.